ShopApp/App/Modules/Main/ViewModels/SplashViewModel.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  SplashViewModel.swift
+//  ShopApp
+//
+//  Created by Evgeniy Antonov on 4/24/18.
+//  Copyright © 2018 RubyGarage. All rights reserved.
+//
+
+import RxSwift
+import ShopApp_Gateway
+
+class SplashViewModel: BaseViewModel {
+    private let cartProductListUseCase: CartProductListUseCase
+    private let cartValidationUseCase: CartValidationUseCase
+    private let deleteCartProductUseCase: DeleteCartProductUseCase
+    
+    var dataLoaded = PublishSubject<Void>()
+    
+    init(cartProductListUseCase: CartProductListUseCase, cartValidationUseCase: CartValidationUseCase, deleteCartProductUseCase: DeleteCartProductUseCase) {
+        self.cartProductListUseCase = cartProductListUseCase
+        self.cartValidationUseCase = cartValidationUseCase
+        self.deleteCartProductUseCase = deleteCartProductUseCase
+    }
+    
+    func loadData() {
+        cartProductListUseCase.getCartProductList { [weak self] (products, error) in
+            guard let strongSelf = self else {
+                return
+            }
+            
+            if let ids = products?.map({ $0.productVariant?.id ?? ""}), !ids.isEmpty, error == nil {
```

Comment:
Instead of `$0.productVariant?.id ?? ""` you can filter all products with $0.productVariant?.id == nil, and after that select non nil ids